sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +50,224 @@ class TableCheckpointStore:
         The hostname of the secondary endpoint.
     """
 
-    def __init__(self, **kwargs):
-        pass
+    def __init__(self, table_account_url, table_name, credential=None, **kwargs):
+        # type(str, str, Optional[Any], Any) -> None
+        self.table_client = kwargs.pop("table_client", None)
+        if not self.table_client:
+            api_version = kwargs.pop("api_version", None)
+            if api_version:
+                headers = kwargs.get("headers")
+                if headers:
+                    headers["x-ms-version"] = api_version
+                else:
+                    kwargs["headers"] = {"x-ms-version": api_version}
+            self.table_client = TableClient(
+                table_account_url, table_name, credential=credential, **kwargs
+            )
+
+    @classmethod
+    def from_connection_string(cls, conn_str, table_name, credential=None, **kwargs):
+        """Create TableCheckpointStore from a storage connection string.
+        :param str conn_str:
+            A connection string to an Azure Storage account.
+        :param table_name:
+            The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+        :keyword str api_version:
+            The Storage API version to use for requests. Default value is '2019-07-07'.
```

Comment:
It looks like the default value is actually '2018-03-28' if we look at the tables package:
https://github.com/Azure/azure-sdk-for-python/blob/d63dcf0b26e6ce6001eb203c7f24594958109644/sdk/tables/azure-data-tables/azure/data/tables/_constants.py#L9